llvm/docs/InstrProfileFormat.rst

```diff
@@ -0,0 +1,476 @@
+are stored contiguously and in an order that is consistent with instrumentation points selection.
+
+.. _calculation-of-counter-offset:
+
+As mentioned above, the recorded counter offset is relative to the profile metadata.
+So how are function counters associated with the profiled function?
+
+Basically, the profile reader iterates profile metadata (from the `profile metadata`_
+section) and makes use of the recorded relative distances, as illustrated below.
+
+::
+
+        + --> start(__llvm_prf_data) --> +---------------------+ ------------+
+        |                                |       Data 1        |             |
+        |                                +---------------------+  =====||    |
+        |                                |       Data 2        |       ||    |
+        |                                +---------------------+       ||    |
+        |                                |        ...          |       ||    |
+ Counter|                                +---------------------+       ||    |
+  Delta |                                |       Data N        |       ||    |
+        |                                +---------------------+       ||    |   CounterPtr1
+        |                                                              ||    |
+        |                                              CounterPtr2     ||    |
+        |                                                              ||    |
+        |                                                              ||    |
+        + --> start(__llvm_prf_cnts) --> +---------------------+       ||    |
+                                         |        ...          |       ||    |
+                                         +---------------------+  -----||----+
+                                         |    Counter for      |       ||
+                                         |       Data 1        |       ||
+                                         +---------------------+       ||
+                                         |        ...          |       ||
+                                         +---------------------+  =====||
+                                         |    Counter for      |
+                                         |       Data 2        |
+                                         +---------------------+
+                                         |        ...          |
+                                         +---------------------+
+                                         |    Counter for      |
+                                         |       Data N        |
+                                         +---------------------+
+
+
+In the graph,
+
+* The profile header records `CounterDelta` with the value as `start(__llvm_prf_cnts) - start(__llvm_prf_data)`.
+  We will call it `CounterDeltaInitVal` below for convenience.
+* For each profile data record, `CounterPtrN` is recorded as `start(Counter) - start(ProfileData)`.
```

Comment:
Counter --> CounterN, ProfileData --> ProfileDataN. 

Also describe that DataN is the N th entry in __llvm_prf_data, and CounterN is the corresponding profile counters.

Reply:
done.